### PR TITLE
Release 0.2.56

### PR DESCRIPTION
**Pull request by johnc1231**
Releases version 0.2.56. Stacked on #9373, since I'm mainly releasing for performance improvements in #9363, #9373, and #9374.

### CODE REVIEW

File: hail/python/hail/docs/change_log.md [hunk 1]
```diff
@@ -22,6 +22,22 @@ Please note that **forward compatibility should not be expected, especially
...
+
+## Performance
+
+- (hail#9373)(hail#9374) Decrease amount of memory used when slicing or filtering along a single BlockMatrix dimension.
+
+## hailctl dataproc
+
+- (hail#9263) Add support for `--expiration-time` argument to `hailctl dataproc start`. 
+- (hail#9263) Add support for `--no-max-idle`, `no-max-age`, `--max-age`, and `--expiration-time` to `hailctl dataproc --modify`.
+
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md:40]: can you add a notice about the bugfix #9304? Maybe: 

```
- (hail#9304) Fix crash in `run_combiner` caused by inputs where VCF lines and BGZ blocks align.
```

Comment (johnc1231) [reply hail/python/hail/docs/change_log.md:44]: Done!

File: hail/python/hail/docs/change_log.md [hunk 2]
```diff
@@ -22,6 +22,26 @@ Please note that **forward compatibility should not be expected, especially
...
+## New features
+
+- (hail#9308) Add hl.enumerate in favor of hl.zip_with_index, which is now deprecated.
+- (hail#9278) Add `ArrayExpression.grouped`, a function that groups hail arrays into fixed size subarrays.
+
+## Performance
+
+- (hail#9373)(hail#9374) Decrease amount of memory used when slicing or filtering along a single BlockMatrix dimension.
+
+## Bug Fixes
```
Comment (danking) [inline hail/python/hail/docs/change_log.md:36]: We always use lower-case fixes.

**Comment by johnc1231** [comment]
Thanks!

**Pull request merged**